internal/model/database.go

```diff
@@ -0,0 +1,281 @@
+package model
+
+//
+// Database
+//
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/upper/db/v4"
+)
+
+// WritabeDatabase supports writing and updating data.
+type WritableDatabase interface {
+	// Session returns the database session
+	Session() db.Session
```

Comment:
Is is possible to omit `Session` from this interface? I would say that perhaps it is possible if the functions that we define below can internally use a `db.Session`. If you think about it, it's difficult otherwise to define a `NullDatabase` that does nothing: what would be return for `db.Session`? We would need to also return a mocked implementation of `db.Session` that does nothing. Conversely, it would be much easier if we could just ignore this function as an implementation detail and omit it from the `WritableDatabase` signature.

One of the reasons why I am advocating for _keeping_ a public `*database.Database` is precisely that code that needs the full `Database` can use the pointer to the correct type, while code that only needs some specific interfaces can use the correct interface. By switching such code to use an interface, we would make it independent of the specific `Database` implementation and thus enable moving it inside `internal`.